File: growing-shop-auth/src/main/java/com/example/growingshopauth/auth/domain/Policy.java
package com.example.growingshopauth.domain;

import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import jakarta.persistence.*;
import java.util.regex.Pattern;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Builder
public class Policy {
    private static final Pattern PATH_REGEX = Pattern.compile("^[\\/]+[\\w-/]+$", Pattern.DOTALL);

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true)
    private String name;

    @Column(nullable = false)
    private String path;

    @Enumerated(EnumType.STRING)
    private HttpMethod method;

    private String description;

    private Policy(Long id, String name, String path, HttpMethod method, String description) {
        validatePath(path);

        this.id = id;
        this.name = name;
        this.path = path;
        this.method = method;
        this.description = description;
    }

    public Policy(String name, String path, HttpMethod method, String description) {
        this(null, name, path, method, description);
    }

    private static void validatePath(String path) {
        if (!PATH_REGEX.matcher(path).matches()) {
            throw new IllegalArgumentException("유효하지 않은 url path 입니다.");
        }
    }
}

File: growing-shop-auth/src/main/kotlin/com/example/growingshopauth/auth/domain/Policy.kt
package com.example.growingshopauth.auth.domain

import com.example.growingshopcommon.entity.BaseEntity
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import java.util.regex.Pattern

@Entity
class Policy(
    @Column(nullable = false, unique = true)
    val name: String,

    @Column(nullable = false)
    val path: String,

    @Enumerated(EnumType.STRING)
    val method: HttpMethod,

    var description: String = ""
) : BaseEntity() {
    companion object {
        private val PATH_REGEX = Pattern.compile("^[\\/]+[\\w-/]+$", Pattern.DOTALL)

        fun validatePath(path: String) {
            if (!PATH_REGEX.matcher(path).matches()) throw IllegalArgumentException("유효하지 않은 url path 입니다.")
        }
    }
}
